Content.Server/Xenoarchaeology/Equipment/Systems/ArtifactAnalyzerSystem.cs

```diff
@@ -0,0 +1,403 @@
+            return;
+
+        if (!Resolve(component.Scanner, ref analyzer))
+            return;
+
+        _audio.PlayPvs(component.ScanFailureSound, component.Scanner, AudioParams.Default.WithVolume(3f));
+
+        RemComp<ActiveArtifactAnalyzerComponent>(component.Scanner);
+        if (analyzer.Console != null)
+            UpdateUserInterface(analyzer.Console.Value);
+
+        RemCompDeferred(artifact, component);
+    }
+
+    /// <summary>
+    /// Finishes the current scan.
+    /// </summary>
+    /// <param name="uid">The analyzer that is scanning</param>
+    /// <param name="component"></param>
+    /// <param name="active"></param>
+    [PublicAPI]
+    public void FinishScan(EntityUid uid, ArtifactAnalyzerComponent? component = null, ActiveArtifactAnalyzerComponent? active = null)
+    {
+        if (!Resolve(uid, ref component, ref active))
+            return;
+
+        component.LastAnalyzedArtifact = active.Artifact;
+        UpdateAnalyzerInformation(uid, component);
+
+        RemComp<ActiveScannedArtifactComponent>(active.Artifact);
+        RemComp(uid, active);
+        if (component.Console != null)
+            UpdateUserInterface(component.Console.Value);
+    }
+
+    private void OnRefreshParts(EntityUid uid, ArtifactAnalyzerComponent component, RefreshPartsEvent args)
+    {
+        var analysisRating = args.PartRatings[component.MachinePartAnalysisDuration];
+
+        component.AnalysisDurationMulitplier = MathF.Pow(component.PartRatingAnalysisDurationMultiplier, analysisRating - 1);
+    }
+
+    private void OnUpgradeExamine(EntityUid uid, ArtifactAnalyzerComponent component, UpgradeExamineEvent args)
+    {
+        args.AddPercentageUpgrade("analyzer-artifact-component-upgrade-analysis", component.AnalysisDurationMulitplier);
+    }
+
+    private void OnCollide(EntityUid uid, ArtifactAnalyzerComponent component, ref StartCollideEvent args)
+    {
+        var otherEnt = args.OtherFixture.Body.Owner;
+
+        if (!HasComp<ArtifactComponent>(otherEnt))
+            return;
+
+        component.Contacts.Add(otherEnt);
+
+        if (component.Console != null)
+            UpdateUserInterface(component.Console.Value);
+    }
+
+    private void OnEndCollide(EntityUid uid, ArtifactAnalyzerComponent component, ref EndCollideEvent args)
+    {
+        var otherEnt = args.OtherFixture.Body.Owner;
+
+        if (!HasComp<ArtifactComponent>(otherEnt))
+            return;
+        component.Contacts.Remove(otherEnt);
+
+        if (component.Console != null)
+            UpdateUserInterface(component.Console.Value);
+    }
+
+    private void OnAnalyzeStart(EntityUid uid, ActiveArtifactAnalyzerComponent component, ComponentStartup args)
+    {
+        if (TryComp<ApcPowerReceiverComponent>(uid, out var powa))
+            powa.NeedsPower = true;
+
+        component.LoopStream = _audio.PlayPvs(component.ScanningSound, uid, AudioParams.Default.WithVolume(3).WithMaxDistance(5).WithLoop(true));
```

Comment:
you should just be using ambient sound for this (i.e. have the ambient sound on the console disabled then just enable it when this starts). otherwise its not going to play for players that come into range afterwards, and the attenuation is going to be wrong (since its just using max distance)